lxd/metrics/types.go

```diff
@@ -0,0 +1,171 @@
+	MemoryShmemBytes
+	// MemorySwapBytes represents the amount of swap memory
+	MemorySwapBytes
+	// MemoryUnevictableBytes represents the amount of unevictable memory
+	MemoryUnevictableBytes
+	// MemoryWritebackBytes represents the amount of memory queued for syncing to disk
+	MemoryWritebackBytes
+	// NetworkReceiveBytesTotal represents the amount of received bytes on a given interface
+	NetworkReceiveBytesTotal
+	// NetworkReceiveDropTotal represents the amount of received dropped bytes on a given interface
+	NetworkReceiveDropTotal
+	// NetworkReceiveErrsTotal represents the amount of received errors on a given interface
+	NetworkReceiveErrsTotal
+	// NetworkReceivePacketsTotal represents the amount of received packets on a given interface
+	NetworkReceivePacketsTotal
+	// NetworkTransmitBytesTotal represents the amount of transmitted bytes on a given interface
+	NetworkTransmitBytesTotal
+	// NetworkTransmitDropTotal represents the amount of transmitted dropped bytes on a given interface
+	NetworkTransmitDropTotal
+	// NetworkTransmitErrsTotal represents the amount of transmitted errors on a given interface
+	NetworkTransmitErrsTotal
+	// NetworkTransmitPacketsTotal represents the amount of transmitted packets on a given interface
+	NetworkTransmitPacketsTotal
+	// ProcsTotal represents the number of running processes
+	ProcsTotal
+)
+
+// MetricNames associates a metric type to its name.
+var MetricNames = map[MetricType]string{
+	CPUSecondsTotal:             "lxd_cpu_seconds_total",
+	DiskReadBytesTotal:          "lxd_disk_read_bytes_total",
+	DiskReadsCompletedTotal:     "lxd_disk_reads_completed_total",
+	DiskWrittenBytesTotal:       "lxd_disk_written_bytes_total",
+	DiskWritesCompletedTotal:    "lxd_disk_writes_completed_total",
+	FilesystemAvailBytes:        "lxd_filesystem_avail_bytes",
+	FilesystemFreeBytes:         "lxd_filesystem_free_bytes",
+	FilesystemSizeBytes:         "lxd_filesystem_size_bytes",
+	MemoryActiveAnonBytes:       "lxd_memory_Active_anon_bytes",
+	MemoryActiveFileBytes:       "lxd_memory_Active_file_bytes",
+	MemoryActiveBytes:           "lxd_memory_Active_bytes",
+	MemoryCachedBytes:           "lxd_memory_Cached_bytes",
+	MemoryDirtyBytes:            "lxd_memory_Dirty_bytes",
+	MemoryHugepagesFreeBytes:    "lxd_memory_HugepagesFree_bytes",
+	MemoryHugepagesTotalBytes:   "lxd_memory_HugepagesTotal_bytes",
+	MemoryInactiveAnonBytes:     "lxd_memory_Inactive_anon_bytes",
+	MemoryInactiveFileBytes:     "lxd_memory_Inactive_file_bytes",
+	MemoryInactiveBytes:         "lxd_memory_Inactive_bytes",
+	MemoryMappedBytes:           "lxd_memory_Mapped_bytes",
+	MemoryMemAvailableBytes:     "lxd_memory_MemAvailable_bytes",
+	MemoryMemFreeBytes:          "lxd_memory_MemFree_bytes",
+	MemoryMemTotalBytes:         "lxd_memory_MemTotal_bytes",
+	MemoryRSSBytes:              "lxd_memory_RSS_bytes",
+	MemoryShmemBytes:            "lxd_memory_Shmem_bytes",
+	MemorySwapBytes:             "lxd_memory_Swap_bytes",
```

Comment:
This represents the total usage of swap, correct?

Reply:
I believe so. We get that from `cg.GetMemorySwapUsage()`.